components/arm/xarm/xarm.go

```diff
@@ -172,7 +171,7 @@ func (x *xArm) Reconfigure(ctx context.Context, deps resource.Dependencies, conf
 		x.conn = newConn
 
 		if err := x.start(ctx); err != nil {
-			return errors.Wrap(err, "failed to start on reconfigure")
+			return errors.Join(err, errors.New("failed to start on reconfigure"))
```

Comment:
[The documentation on `Join`](https://pkg.go.dev/errors#Join) reminds me more of [`multierr.Combine`](https://pkg.go.dev/go.uber.org/multierr#Combine) than [the old `Wrap`](https://pkg.go.dev/github.com/pkg/errors#Wrap). Specifically, `Join` will put these errors on separate lines (as though they're separate errors), whereas `Wrap` would put them on the same line with a colon in between (as though `err` provides extra detail on why we failed to start on reconfigure). 

Looking at [the documentation on `Unwrap`](https://pkg.go.dev/errors#Unwrap), I wonder if `fmt.Errorf("failed to start on reconfigure: %w", err)` mirrors the old behavior more closely.

edit: I wrote that, and then found https://github.com/kubernetes/release/issues/2549 saying roughly the same thing. Similarly, https://github.com/xdg-go/go-rewrap-errors looks like it's intended to rewrite your code for you with this same transformation, as described in https://xdg.me/rewriting-go-with-ast-transformation/.

Reply:
Help me out, would that look like:
``` fmt.Errorf("%w, something something %s", originalErr, "somestring" ```

Reply:
I think more like `fmt.Errorf("failed to start on %s: %w", "reconfigure", originalErr)`. That way, the final result is "failed to start on reconfigure: cannot open file: permission denied" or similar, just like the old `errors.Wrap` did.

I've been reading https://xdg.me/rewriting-go-with-ast-transformation/, and it sure looks like that article has code that will automatically make this transformation for us, rather than doing it manually.

Reply:
I'll try that next, for now I have to go back to my regularly scheduled programming :D